java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestGenerationUtils.java

```diff
@@ -264,19 +266,13 @@ private static TextEdit getTextEdit(TestKind kind, List<MethodMetaData> methodMe
         for (final MethodMetaData method : methodMetadata) {
             
             final MethodDeclaration decl = ast.newMethodDeclaration();
-            // JUnit 4's test method must be public
-            if (kind == TestKind.JUnit) {
-                decl.modifiers().addAll(ASTNodeFactory.newModifiers(ast, Modifier.PUBLIC));
-            }
-
-            // @BeforeClass and @AfterClass in JUnit 4 & 5 needs static modifier
-            if (needStaticModifier(kind, method.annotation)) {
-                decl.modifiers().addAll(ASTNodeFactory.newModifiers(ast, Modifier.STATIC));
-            }
 
             // set a unique method name according to the annotation type
-            decl.setName(ast.newSimpleName(getUniqueMethodName(typeBinding.getJavaElement(),
-                    method.methodName)));
+            final String methodName = getUniqueMethodName(typeBinding.getJavaElement(), method.methodName);
```

Comment:
When finding unique method names, should also filter out the method names in super class with different signature.